File: ConicBundle/cppinterface/cb_nncboxsupportfunction.cpp
dll void cb_nncboxsupportfunction_destroy(NNCBoxSupportFunction* self) {
  delete self;
}

dll NNCBoxSupportFunction* cb_nncboxsupportfunction_new(const Matrix* lb, const Matrix* ub, int incr = -1) {
  return new NNCBoxSupportFunction(*lb, *ub, 0, incr);
}

dll int cb_nncboxsupportfunction_check_correctness(const NNCBoxSupportFunction* self) {
  return self->check_correctness();
}

dll const Matrix* cb_nncboxsupportfunction_get_lower_bounds(NNCBoxSupportFunction* self) {
  return &self->get_lower_bounds();
}

dll const Matrix* cb_nncboxsupportfunction_get_upper_bounds(NNCBoxSupportFunction* self) {
  return &self->get_upper_bounds();
}

dll void cb_nncboxsupportfunction_set_out(NNCBoxSupportFunction* self, int pril = 1) {
  self->set_out(&std::cout, pril);
}

dll void cb_nncboxsupportfunction_set_cbout(NNCBoxSupportFunction* self, int incr = -1) {
  self->set_cbout(0, incr);
}

dll void cb_nncboxsupportfunction_print_problem_data(const NNCBoxSupportFunction* self) {
  self->print_problem_data(std::cout);
}

dll void cb_nncboxsupportfunction_print_problem_data_to_mfile(const NNCBoxSupportFunction* self, Integer blocknr) {
  self->print_problem_data_to_mfile(std::cout, blocknr);
}

File: src/cppinterface/cb_nncboxsupportfunction.jl
@doc raw"""
    CBNNCBoxSupportFunction(lb::CBMatrix, ub::CBMatrix, incr::Integer = -1)

intialize with lower bounds vector lb and upper bounds vector ub (and output options), both must be column vectors of the same length and lb<=ub componentwise, length 0 results in objective value 0
"""
CBNNCBoxSupportFunction(lb::CBMatrix, ub::CBMatrix, incr::Integer = -1) = CBNNCBoxSupportFunction(@ccall libcb.cb_nncboxsupportfunction_new(lb.data::Ptr{Cvoid}, ub.data::Ptr{Cvoid}, incr::Cint)::Ptr{Cvoid})

@doc raw"""
    cb_check_correctness(self::CBNNCBoxSupportFunction)

see MatrixFunctionOracle::check_correctness()   (true only needed for debugging)
"""
cb_check_correctness(self::CBNNCBoxSupportFunction) = Bool(@ccall libcb.cb_nncboxsupportfunction_check_correctness(self.data::Ptr{Cvoid})::Cint)

@doc raw"""
    cb_get_lower_bounds!(self::CBNNCBoxSupportFunction)

returns the column vector of lower bounds
"""
cb_get_lower_bounds!(self::CBNNCBoxSupportFunction) = (@ccall libcb.cb_nncboxsupportfunction_get_lower_bounds(self.data::Ptr{Cvoid})::Ptr{Cvoid}; return self)

@doc raw"""
    cb_get_upper_bounds!(self::CBNNCBoxSupportFunction)

retunrs the column vector of upper bounds
"""
cb_get_upper_bounds!(self::CBNNCBoxSupportFunction) = (@ccall libcb.cb_nncboxsupportfunction_get_upper_bounds(self.data::Ptr{Cvoid})::Ptr{Cvoid}; return self)

@doc raw"""
    cb_set_out!(self::CBNNCBoxSupportFunction, pril::Integer = 1)

see ConicBundle::CBout
"""
cb_set_out!(self::CBNNCBoxSupportFunction, pril::Integer = 1) = @ccall libcb.cb_nncboxsupportfunction_set_out(self.data::Ptr{Cvoid}, pril::Cint)::Cvoid

@doc raw"""
    cb_set_cbout!(self::CBNNCBoxSupportFunction, incr::Integer = -1)

see ConicBundle::CBout
"""
cb_set_cbout!(self::CBNNCBoxSupportFunction, incr::Integer = -1) = @ccall libcb.cb_nncboxsupportfunction_set_cbout(self.data::Ptr{Cvoid}, incr::Cint)::Cvoid

@doc raw"""
    cb_print_problem_data(self::CBNNCBoxSupportFunction)

write the problem description to out so that it can be read again by read_problem_data()
"""
cb_print_problem_data(self::CBNNCBoxSupportFunction) = @ccall libcb.cb_nncboxsupportfunction_print_problem_data(self.data::Ptr{Cvoid})::Cvoid

@doc raw"""
    cb_print_problem_data_to_mfile(self::CBNNCBoxSupportFunction, blocknr::Integer)

undocumented highly volatile variant for external testing
"""
cb_print_problem_data_to_mfile(self::CBNNCBoxSupportFunction, blocknr::Integer) = @ccall libcb.cb_nncboxsupportfunction_print_problem_data_to_mfile(self.data::Ptr{Cvoid}, blocknr::Cint)::Cvoid

